### PR TITLE
follow-up test_torch .data removal

### PR DESCRIPTION
**Pull request by albanD**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#33696 follow-up test_torch .data removal**

This changes two tests:
- The batchnorm inference cannot change the memory format of the weights as they are 1D. So this is removed.
- The batchnorm test now run both in affine and not affine mode.
- I added back the test for type errors using .data. In particular, `.data` allows to change the type of a Tensor inplace (very bad, never do it!) but since it is possible, we should test it until .data is removed.

cc @Enealor who did the first version of the PR.

Differential Revision: [D20069241](https://our.internmc.facebook.com/intern/diff/D20069241)

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -5277,26 +5277,23 @@ def test_batch_norm_cpu_inference(self):
...
+                m = torch.nn.BatchNorm2d(inputs[i].size()[1], 1e-05, 0.1, affine=affine)
+                m.eval()
+                # contiguous case
+                input1 = inputs[i].contiguous()
+                output1 = m(input1)
+                # non-contiguous case
+                input2 = input1.permute(0, 1, 3, 2)
+                output2 = m(input2).permute(0, 1, 3, 2)
+                # channels last case
+                input3 = input1.contiguous(memory_format=torch.channels_last)
```
Comment (gchanan) [inline test/test_torch.py]: you also need to change the parameters to make sure we get channels_last algorithms.

Comment (albanD) [reply test/test_torch.py]: The only parameters in the affine case are two 1D Tensors of size number of features. So they cannot be changed to channels_last.
The running stats are 1D Tensors so we cannot change them either.

Which parameters do you have in mind?

Comment (gchanan) [reply test/test_torch.py]: oh you are right, and I pointed out something similar in the original review :).

**Review by gchanan — APPROVED** [review]

File: test/test_torch.py (hunk 1 above)
Comment (albanD) [reply test/test_torch.py]: Yes exactly, I linked your comment in the original removal PR.
